File: backend/Wms.Api/Domain/Entities/StockTransfer.cs
namespace Wms.Api.Domain.Entities;

public class StockTransfer
{
    public int Id { get; set; }
    public required string TransferNumber { get; set; }
    public int FromLocationId { get; set; }
    public int ToLocationId { get; set; }
    public DateTime TransferredAtUtc { get; set; } = DateTime.UtcNow;
    public Location FromLocation { get; set; } = null!;
    public Location ToLocation { get; set; } = null!;
    public ICollection<StockTransferLine> Lines { get; set; } = [];
}

File: frontend/wms-web/src/types/stockTransfer.ts
export type StockTransferLine = {
  id: number
  productId: number
  productSku: string
  productName: string
  quantity: number
}

export type StockTransfer = {
  id: number
  transferNumber: string
  fromLocationId: number
  fromLocationCode: string
  fromWarehouseName: string
  toLocationId: number
  toLocationCode: string
  toWarehouseName: string
  transferredAtUtc: string
  lines: StockTransferLine[]
}

export type CreateStockTransferRequest = {
  fromLocationId: number
  toLocationId: number
  lines: Array<{ productId: number; quantity: number }>
}
